components/locale_core/src/extensions/unicode/value.rs

```diff
@@ -102,6 +103,31 @@ impl Value {
         }
     }
 
+    /// A constructor which takes a pre-sorted list of [`Value`] elements.
+    ///
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Value;
+    /// use tinystr::{TinyAsciiStr, tinystr};
+    ///
+    /// let tinystr1: TinyAsciiStr<8> = tinystr!(8, "foobar");
+    /// let tinystr2: TinyAsciiStr<8> = tinystr!(8, "testing");
+    /// let mut v = vec![tinystr1, tinystr2];
+    /// v.sort();
+    /// v.dedup();
+    ///
+    /// let value = Value::from_vec_unchecked(v);
+    /// ```
+    ///
+    /// Notice: For performance- and memory-constrained environments, it is recommended
+    /// for the caller to use [`binary_search`](slice::binary_search) instead of [`sort`](slice::sort)
+    /// and [`dedup`](Vec::dedup()).
+    pub fn from_vec_unchecked(input: Vec<TinyAsciiStr<8>>) -> Self {
+        Self(input.into())
+    }
```

Comment:
Issue: I think this involves reallocating, so there's no advantage to taking a `Vec` over a slice.

Reply:
I don't think it reallocates, see https://github.com/unicode-org/icu4x/blob/0ab2630363924ca87aafc9f24f74c4b73cac60ec/components/locale_core/src/shortvec/mod.rs#L248 

This is used in Attributes and Value. If you want, I can add `From<&[T]>` to ShortBoxSlice and then switch this to be from _slice_unchecked, or use from_short_slice_unchecked` on both. Lmk.

Reply:
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.shrink_to_fit

> The behavior of this method depends on the allocator, which may either shrink the vector in-place or reallocate. The resulting vector might still have some excess capacity, just as is the case for with_capacity. See Allocator::shrink for more details.

So in certain situations there's an advantage to taking a vec.

Reply:
We already have from_tinystr(subtag) so there's already a fast path, so this other function can be used only when multiple subtags are needed. Seems fine